### PR TITLE
Mice no longer spawn on grilles

### LINKED ISSUE #21574: Mice spawning in space
5:20 pm EST 11/15/16, Bagil. Map is metastation, a mouse spawned in space near the AI upload.
home trade post xxx 2016-11-15 172140

### PR DESCRIPTION
**Pull request by coiax**
Fixes #21574. Mice no longer spawn on cables that have dense objects
also on the tile.

Also added the Mice Migration event, used for debug purposes, currently
disabled.

### CODE REVIEW

File: code/controllers/subsystem/squeak.dm [hunk 1]
```diff
@@ -35,6 +37,10 @@ var/datum/subsystem/squeak/SSsqueak
...
 
 	var/list/all_turfs = block(locate(1,1,1), locate(world.maxx,world.maxy,1))
-	for(var/turf/open/floor/plating/T in all_turfs)
-		if(locate(/obj/structure/cable) in T)
-			exposed_wires += T
+	turf_loop:
+		for(var/turf/open/floor/plating/T in all_turfs)
+			if(locate(/obj/structure/cable) in T)
+				for(var/atom/A in T)
+					if(A.density)
```
Comment (ChangelingRain) [inline code/controllers/subsystem/squeak.dm]: there's a proc for this!! `is_blocked_turf(turf_in_question)`, returns TRUE if turf is blocked

**Comment by PKPenguin321** [comment]
Also make them not space proof

**Comment by Supermichael777** [comment]
SOUNDS LIKE SOMTHING TO ADD TO CHRISTMASS

**Comment by lordpidey** [comment]
You need to add max_occurances to 0 (or something, I forget exact variable name), since 0 weight isn't enough to totally disable it for some reason.

That said, if it works as advertised, I see no reason for the event to be totally removed.

**Pull request merged**